rtos/rtx2/TARGET_CORTEX_M/rt_OsEventObserver.c

```diff
@@ -0,0 +1,51 @@
+/*
```

Comment:
No, we haven't proposed `OsEventObserver` to upstream, yet. The only client of `OsEventObserver` is uVisor, which doesn't have a Cortex A version.

RTX5 has new tracing hooks that uVisor may be able to use, but we haven't had the time to invest in looking into this. We'd want to look into using the new tracing hooks before proposing to upstream this. It wouldn't make sense to propose a new feature when another already existing feature exists. Until we have the time to look into using the new tracing hooks, unfortunately, uVisor depends on mbed OS providing `OsEventObserver` separately.

The main difference between the new tracing hooks and the `OsEventObserver` is that `OsEventObserver` runs the hooks in privileged mode. uVisor uses the `OsEventObserver` to, among a handful of other things, switch boxes on thread switch when switching between threads owned by different boxes. The `OsEventObserver` hooks are called in very specific places to ensure that box switching happens not too soon and not too late in the switching process (as thread stacks are often private to a box). The new tracing hooks appear unsuitable for such a careful placement at first glance, but a more careful looking into is required.